cmd/juju/application/deploy.go

```diff
@@ -249,16 +275,69 @@ func NewDeployCommandForTest(newAPIRoot func() (DeployAPI, error), steps []Deplo
 				charmsClient:      &charmsClient{Client: apicharms.NewClient(apiRoot)},
 				applicationClient: &applicationClient{Client: application.NewClient(apiRoot)},
 				modelConfigClient: &modelConfigClient{Client: modelconfig.NewClient(apiRoot)},
-				charmstoreClient:  &charmstoreClient{Client: cstoreClient},
+				charmstoreClient:  &charmstoreClient{&charmstoreClientShim{cstoreClient}},
 				annotationsClient: &annotationsClient{Client: annotations.NewClient(apiRoot)},
-				charmRepoClient:   &charmRepoClient{CharmStore: charmrepo.NewCharmStoreFromClient(cstoreClient)},
+				charmRepoClient:   &charmRepoClient{charmrepo.NewCharmStoreFromClient(cstoreClient)},
 				plansClient:       &plansClient{planURL: mURL},
 			}, nil
 		}
 	}
 	return modelcmd.Wrap(deployCmd)
 }
 
+// NewDeployCommandForTest2 returns a command to deploy applications intended to be used only in tests
+// that do not use gomock.
+func NewDeployCommandForTest2(charmstore charmstoreForDeploy, charmrepo charmrepoForDeploy) modelcmd.ModelCommand {
```

Comment:
This should be in `export_test.go`

Reply:
`NewDeployCommandForTest` also lives in that file. Added this function in there for consistency. Should I move both in?

Reply:
any FooForTest methods normally go in export_test

Reply:
Sorted. Have added both in there